core/src/main/java/org/apache/accumulo/core/iterators/Combiner.java

```diff
@@ -149,6 +159,37 @@ public void next() throws IOException {
 
   private Key workKey = new Key();
 
+  private static Cache<String,Long> loggedMsgCache = CacheBuilder.newBuilder().expireAfterWrite(1, TimeUnit.HOURS).build();
+
+  private void sawDelete() {
+    if (isPartialCompaction) {
+      switch (deleteHandlingAction) {
+        case LOG_ERROR:
+          try {
+            loggedMsgCache.get(this.getClass().getName(), new Callable<Long>() {
+              @Override
+              public Long call() throws Exception {
+                log.error("Combiner of type " + this.getClass().getSimpleName()
+                    + " saw a delete during a partial compaction.  This could cause undesired results.  See ACCUMULO-2232.  Will not log subsequent occurences for at least 1 hour.");
+                return System.currentTimeMillis();
```

Comment:
Do we really need the current time? Isn't `0` sufficient since we don't inspect the value of the cache entries? A comment that the value returned by the Callable is meaningless would also be nice.